src/nodes.coffee

```diff
@@ -1591,6 +1604,12 @@ exports.Code = class Code extends Base
     @bound       = tag is 'boundfunc'
     @isGenerator = !!@body.contains (node) ->
       (node instanceof Op and node.isYield()) or node instanceof YieldReturn
+    @isAsync     = !!@body.contains (node) =>
+      if (node instanceof Op and node.isAwait()) or node instanceof AwaitReturn
+        node.error "generator function can't contain await" if @isGenerator
```

Comment:
@GeoffreyBooth That won't work because `node` isn't visible in that scope, but I share your concern about the asymmetry of `await` and `yield`. I will fix that, but I still think its best to point users toward a specific line of code rather than an entire function.